bigquery/google/cloud/bigquery/job.py

```diff
@@ -542,35 +540,153 @@ def cancelled(self):
                 and self.error_result.get('reason') == _STOPPED_REASON)
 
 
-class _LoadConfiguration(object):
-    """User-settable configuration options for load jobs.
+class LoadJobConfig(object):
+    """Configuration options for load jobs.
 
-    Values which are ``None`` -> server defaults.
+    All properties in this calss are optional. Values which are ``None`` ->
+    server defaults.
     """
-    _allow_jagged_rows = None
-    _allow_quoted_newlines = None
-    _autodetect = None
-    _create_disposition = None
-    _encoding = None
-    _field_delimiter = None
-    _ignore_unknown_values = None
-    _max_bad_records = None
-    _null_marker = None
-    _quote_character = None
-    _skip_leading_rows = None
-    _source_format = None
-    _write_disposition = None
+
+    def __init__(self):
+        self._properties = {}
+        self._schema = ()
```

Comment:
I did something different for QueryJobConfig (not quite ready to mail yet, unfortunately)

https://github.com/tswast/google-cloud-python/blob/da1469f9d421df559d27150b39ce6e9932b65c6b/bigquery/google/cloud/bigquery/job.py#L1355

Basically, I stored the nested properties as the native objects in `_properties` but then do some conversion in the to/from API repr methods.

You might want to do something similar. Either way, you're right that we do need to treat the schema property differently.

Reply:
Given that I have to treat setting a schema specially anyway, and there's only that and skipLeadingRows as special cases, I'm not sure that the extra machinery in that CL is worth it. Can we leave it like this for now?